akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.Graph
+import akka.stream.BidiShape
+import akka.stream.impl.StreamLayout.Module
+import akka.stream.FlowShape
+
+final class BidiFlow[-I1, +O1, -I2, +O2, +Mat](private[stream] override val module: Module) extends Graph[BidiShape[I1, O1, I2, O2], Mat] {
```

Comment:
We probably need a factory from a Graph[BidiShape, _]. The same should be added to Flow, Source and Sink (maybe RunnableFlow, too, but that is less useful).